test/bin/scenario_container.sh

```diff
@@ -0,0 +1,162 @@
+#!/bin/bash
+
+if [[ "${BASH_SOURCE[0]}" == "${0}" ]]; then
+    echo "This script must be sourced, not executed."
+    exit 1
+fi
+
+process_kickstart_into_script() {
+    local -r container_name="$1"
+    local -r file=${2:-kickstart.ks}
+
+    local -r output_dir="${SCENARIO_INFO_DIR}/${SCENARIO}/vms/${container_name}"
+
+    while IFS= read -r line; do
+        if [[ $line =~ ^%include[[:space:]]+(.+) ]]; then
+            process_kickstart_into_script "${container_name}" "${BASH_REMATCH[1]}"
+        else
+            echo "$line" >> "${output_dir}/init.sh"
+        fi
+    done < "${output_dir}/${file}"
+}
+
+create_container_entrypoint() {
+    local -r name="$1"
+    local -r output_dir="${SCENARIO_INFO_DIR}/${SCENARIO}/vms/${name}/"
+
+    cat << EOF > "${output_dir}/init.sh"
+#!/bin/bash
+set -xeuo pipefail
+
+cat /proc/sys/kernel/random/uuid > /proc/sys/kernel/random/boot_id
+
+if [ -e /var/container-init-ok ]; then
+    echo "Container already initialized"
+    exit 0
+fi
+
+EOF
+    chmod +x "${output_dir}/init.sh"
+    process_kickstart_into_script "${name}"
+
+    cat << EOF >> "${output_dir}/init.sh"
+touch /var/container-init-ok
+
+EOF
+
+    cat << 'EOF' > "${output_dir}/entrypoint.sh"
+#!/bin/bash
+SCRIPTDIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+bash -x "${SCRIPTDIR}/init.sh" &> /var/log/container_init.log
+exec /sbin/init
+EOF
+    chmod +x "${output_dir}/entrypoint.sh"
+}
+
+launch_container() {
+    local name="host1"
+    local image=""
+    local memory=4096
+    local vcpus=2
+
+    while [ $# -gt 0 ]; do
+        case "$1" in
+            --name|--image|--vcpus|--memory)
+                var="${1/--/}"
+                if [ -n "$2" ] && [ "${2:0:1}" != "-" ]; then
+                    declare "${var}=$2"
+                    shift 2
+                else
+                    error "Failed parsing arguments: ${var} value not set"
+                    record_junit "${name}" "ctr-launch-args" "FAILED"
+                    exit 1
+                fi
+                ;;
+            *)
+                error "Invalid argument: ${1}"
+                record_junit "${name}" "ctr-launch-args" "FAILED"
+                exit 1
+                ;;
+        esac
+    done
+
+    if [ -z "${image}" ]; then
+        echo "--image is empty"
+        exit 1
+    fi
+
+    local -r full_image="${MIRROR_REGISTRY_URL}/${image}"
+
+    record_junit "${name}" "ctr-launch-args" "OK"
+
+    local -r full_ctr_name=$(full_vm_name "${name}")
+
+    if sudo podman inspect -f "{{.Id}}" "${full_ctr_name}" &> /dev/null; then
+        echo "Container '${full_ctr_name}' already exists."
```

Comment:
Should this be an error to be consistent with startup of VMs?

Reply:
What do you mean? Remove "Container" so it's exactly the same?
```
    if sudo virsh dominfo "${full_vmname}" 2>/dev/null; then
        echo "${full_vmname} already exists"
        record_junit "${vmname}" "install_vm" "SKIP"
        return 0
    fi
```

Reply:
Ah, I see [here](https://github.com/openshift/microshift/blob/main/test/bin/scenario.sh#L503-L507) that we're skipping VM creation the same way. All good.